src/serve_session.rs

```diff
@@ -190,7 +219,10 @@ impl ServeSession {
     }
 
     pub fn project_name(&self) -> &str {
-        &self.root_project.name
+        self.root_project
+            .name
+            .as_ref()
+            .expect("all top-level projects must have their name set")
```

Comment:
Is this panic reachable?

Reply:
It shouldn't be. I gave it a separate warning just in case though.

Reply:
Cool, I'm approving this then and we'll have @osyrisrblx check out #871 before merging since that's the one that's shipping right now